features/step_definitions/barclamp_rabbitmq/rabbitmq_steps.rb

```diff
@@ -0,0 +1,74 @@
+Given(/^the barclamp proposal is using rabbitmq as messaging backend$/) do
+    json_response = JSON.parse(admin_node.exec!("crowbar rabbitmq show default").output)
+    expect(json_response["attributes"]["rabbitmq"]).not_to be_empty
+end
+
+When(/^the node with rabbitmq-server role has been detected successfully$/) do
+    json_response = JSON.parse(admin_node.exec!("crowbar rabbitmq show default").output)
+    @node_fqdn = json_response["deployment"]["rabbitmq"]["elements"]["rabbitmq-server"].first
+    expect(@node_fqdn).not_to be_empty
+end
+
+Then(/^I can check the status of rabbitmq-server using command line$/) do
+    rabbitmq_status = admin_node.exec!("rabbitmqctl status").output.lines[1..-1] 
```

Comment:
yes. It would be connecting to the server on admin node. So I now modified to use the search method to find the appropriate node based on role. This should make sure I am testing on the correct node. This also works for HA setup.